eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/TxPublisher.scala

```diff
@@ -196,22 +194,21 @@ private class TxPublisher(nodeParams: NodeParams, factory: TxPublisher.ChildFact
         }
 
       case cmd: PublishReplaceableTx =>
-        val targetFeerate = feeEstimator.getFeeratePerKw(feeTargets.commitmentBlockTarget)
         val attempts = pending.getOrElse(cmd.input, Seq.empty)
-        val alreadyPublished = attempts.exists {
-          // If there is already an attempt at spending this outpoint with a higher feerate, there is no point in publishing again.
-          case a: ReplaceableAttempt => targetFeerate <= a.feerate
-          case _ => false
+        val alreadyPublished = attempts.collectFirst {
+          // If there is already an attempt at spending this outpoint with a more aggressive confirmation target, there is no point in publishing again.
+          case a: ReplaceableAttempt if a.cmd.txInfo.confirmBefore <= cmd.txInfo.confirmBefore => a.cmd.txInfo.confirmBefore
         }
-        if (alreadyPublished) {
-          log.info("not publishing replaceable {} spending {}:{} with feerate={}, publishing is already in progress", cmd.desc, cmd.input.txid, cmd.input.index, targetFeerate)
-          Behaviors.same
-        } else {
-          val publishId = UUID.randomUUID()
-          log.info("publishing replaceable {} spending {}:{} with id={} ({} other attempts)", cmd.desc, cmd.input.txid, cmd.input.index, publishId, attempts.length)
-          val actor = factory.spawnReplaceableTxPublisher(context, TxPublishLogContext(publishId, channelInfo.remoteNodeId, channelInfo.channelId_opt))
-          actor ! ReplaceableTxPublisher.Publish(context.self, cmd, targetFeerate)
-          run(pending + (cmd.input -> attempts.appended(ReplaceableAttempt(publishId, cmd, targetFeerate, actor))), retryNextBlock, channelInfo)
+        alreadyPublished match {
+          case Some(currentConfirmationTarget) =>
+            log.info("not publishing replaceable {} spending {}:{} with confirmation target={}, publishing is already in progress with confirmation target={}", cmd.desc, cmd.input.txid, cmd.input.index, cmd.txInfo.confirmBefore, currentConfirmationTarget)
+            Behaviors.same
+          case None =>
+            val publishId = UUID.randomUUID()
+            log.info("publishing replaceable {} spending {}:{} with id={} ({} other attempts)", cmd.desc, cmd.input.txid, cmd.input.index, publishId, attempts.length)
+            val actor = factory.spawnReplaceableTxPublisher(context, TxPublishLogContext(publishId, channelInfo.remoteNodeId, channelInfo.channelId_opt))
```

Comment:
Why do you choose to spawn a new attempt instead of just bumping the existing one, which would remove one level of complexity? Is it because the output of the transaction might be different? But does that really matter?

Reply:
This is good feedback, we should update this now that we have a fee bumping mechanism.
Done in https://github.com/ACINQ/eclair/pull/2113/commits/4e635defaf90dde317e00bf6833658cb6c5890ac